raft/log_test.go

```diff
@@ -281,9 +281,7 @@ func TestLog_WriteEntriesTo_Snapshot(t *testing.T) {
 	}()
 
 	// Wait for buffer to be written to.
-	for buf.Len() == 0 {
-		runtime.Gosched()
-	}
+	time.Sleep(1 * time.Second)
```

Comment:
Thanks @vaskoz 

OK, I see what you've done by adding a lock to `Read` as well i.e. `Read` has been locked. With that lock in place, do we still need to replace `runtime.Gosched`? If so, why? 

This is an important change, as it may fix some issues I also have seen in unit testing, and I'm keen to fix it. That's why I want us both to understand this change.